os/mm/mm_heap/mm_partition_mgr.c

```diff
@@ -136,13 +159,31 @@ int8_t mm_allocate_ram_partition(uint32_t **start_addr, uint32_t *size)
 		*size = g_default_size;
 	}
 
+#ifdef CONFIG_MM_PARTITION_HEAP
+
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
 	ARCH_GET_RET_ADDRESS
 	*start_addr = mm_malloc(&g_pheap, *size + sizeof(struct mm_heap_s), retaddr);
 #else
 	*start_addr = mm_malloc(&g_pheap, *size + sizeof(struct mm_heap_s));
 #endif
 
+#else	/* CONFIG_MM_PARTITION_HEAP */
+	uint32_t overhead = sizeof(struct mm_heap_s);
+	if (g_mm_parts[0].size == 0 && g_mm_parts[1].start > g_ram_start + *size) {
+		*start_addr = (uint32_t *)g_ram_start;
+		g_mm_parts[0].start = *start_addr;
+		g_mm_parts[0].size = *size + overhead;
+	} else if (g_mm_parts[1].size == 0 && g_ram_start + g_ram_size - *size > g_ram_start + g_mm_parts[0].size) {
```

Comment:
You assigned the `*start_addr` to `g_ram_start + g_ram_size - *size - overhead`.
Then I think, `g_ram_start + g_ram_size - *size > g_ram_start + g_mm_parts[0].size` this comparison should be changed to
`g_ram_start + g_ram_size - *size - overhead > g_ram_start + g_mm_parts[0].size`.

Reply:
Done.